ext/OptimizationSparseFinitediffExt.jl

```diff
@@ -0,0 +1,243 @@
+module OptimizationSparseFinitediffExt
+
+import SciMLBase: OptimizationFunction
+import Optimization, ArrayInterface
+import ADTypes: AutoFiniteDiff
+import Symbolics
+using LinearAlgebra
+isdefined(Base, :get_extension) ? (using FiniteDiff, SparseDiffTools) :
+(using ..FiniteDiff, ..SparseDiffTools)
+
+const FD = FiniteDiff
+
+function Optimization.instantiate_function(f, x, adtype::AutoFiniteDiff, p,
```

Comment:
won't this throw a warning about overriding?

Reply:
Yes it does...